worlds/oot_soh/location_access/overworld/zoras_domain.py

```diff
@@ -4,6 +4,143 @@
     from ... import SohWorld
 
 
+class EventLocations(str, Enum):
+    ZD_GOSSIP_STONE = "ZD Gossip Stone"
+    ZD_NUT_POT = "ZD Nut Pot"
+    ZD_STICK_POT = "ZD Stick Pot"
+    ZD_FISH_GROUP = "ZD Fish Group"
+    ZD_KING_ZORA_THAWED = "ZD King Zora Thawed"
+    ZD_DELIVER_RUTOS_LETTER = "ZD Deliver Ruto's Letter"
+    ZD_STORMS_GROTTO_FAIRY_ACCESS = "ZD STORMS GROTTO FAIRY ACCESS"
+
+
+class LocalEvents(str, Enum):
+    KING_ZORA_THAWED = "King Zora Thawed"
+
+
 def set_region_rules(world: "SohWorld") -> None:
-    player = world.player
-    
+    # Zoras Domain
+    # Events
+    add_events(Regions.ZORAS_DOMAIN, world, [
+        (EventLocations.ZD_GOSSIP_STONE, Events.CAN_ACCESS_FAIRIES,
+         lambda bundle: call_gossip_fairy_except_suns(bundle)),
+        (EventLocations.ZD_NUT_POT, Events.CAN_FARM_NUTS, lambda bundle: True),
+        (EventLocations.ZD_STICK_POT, Events.CAN_FARM_STICKS, lambda bundle: is_child(bundle)),
+        (EventLocations.ZD_FISH_GROUP, Events.CAN_ACCESS_FISH, lambda bundle: is_child(bundle)),
+        (EventLocations.ZD_KING_ZORA_THAWED, LocalEvents.KING_ZORA_THAWED,
+         lambda bundle: is_adult(bundle) and blue_fire(bundle)),
```

Comment:
So based on the previous comment, you could call this EventLocation `ZORAS_DOMAIN_KING_ZORA_THAWING`